model/riscv_insts_vext_vset.sail

```diff
@@ -65,48 +60,46 @@ function clause execute VSET_TYPE(op, ma, ta, sew, lmul, rs1, rd) = {
     VSETVLI => {
       let vtype_reserved = zeros(sizeof(xlen) - 9);
       let vtype_val : xlenbits = 0b0 @ vtype_reserved @ ma @ ta @ sew @ lmul;
-      writeCSR(csr_name_map("vtype"), vtype_val);
-      let LMUL_new  : real = get_vtype_LMUL();
-      let SEW_new   : int = get_vtype_vsew();
-      VLMAX = floor(LMUL_new) * vlen / SEW_new /* VLMAX = LUML * VLEN / SEW */
+      writeCSR(0xC21, vtype_val)
```

Comment:
Still no, like I said, don't use writeCSR here, it is not for this

Reply:
I'm using `writeCSR` for the convenience of log printing. If we don't care about logs, I can definitely use the direct assignment statement. But if we do, and I should write a separate new function for printing logs for vector CSRs, why not just use the existing `writeCSR` function? Why `writeCSR` not for vector CSRs?

Reply:
Because it forces you to provide an index and unnecessarily requires extending the value to XLEN bits only to have it truncated internally again. The problems are:
1. You can get the index wrong and not notice until run time
2. The index is a magic number that's not as readable
3. If you set high bits outside the register's bit width then they get silently truncated rather than having the type checker tell you you've screwed up
writeCSR is an internal helper function for the implementation of CSRR[WSC](I). Do not use it for other things.

Also the logging it provides would be confusing to print, anyway, as you wouldn't want the "(input: ...)" it tacks on the end, since there is no user input for the value, it's implied.

Reply:
Makes sense very much. It seems that writing a helper function like this cannot avoid the problems of passing the CSR index and extend the value length. So in this part of code, maybe directly using "print" when needed without a helper function is the better way?

Reply:
vset instructions are modified